File: src/docs/Pages/FontSize.fs
module Docs.Pages.FontSize

open Feliz
open Elmish
open Feliz.UseElmish
open Feliz.DaisyUI
open Feliz.Tailwind
open Feliz.Tailwind.Operators
open Docs.SharedView

let fontSizeStyles = [
    fontSize.textXs
    fontSize.textSm
    fontSize.textBase
    fontSize.textLg
    fontSize.textXl
    fontSize.text2Xl
    fontSize.text3Xl
    fontSize.text4Xl
    fontSize.text5Xl
    fontSize.text6Xl
    fontSize.text7Xl
    fontSize.text8Xl
    fontSize.text9Xl
]

let str prop =
    if prop = fontSize.textXs then "text-xs"
    elif prop = fontSize.textSm then "text-sm"
    elif prop = fontSize.textBase then "text-base"
    elif prop = fontSize.textLg then "text-lg"
    elif prop = fontSize.textXl then "text-xl"
    elif prop = fontSize.text2Xl then "text-2xl"
    elif prop = fontSize.text3Xl then "text-3xl"
    elif prop = fontSize.text4Xl then "text-4xl"
    elif prop = fontSize.text5Xl then "text-5xl"
    elif prop = fontSize.text6Xl then "text-6xl"
    elif prop = fontSize.text7Xl then "text-7xl"
    elif prop = fontSize.text8Xl then "text-8xl"
    elif prop = fontSize.text9Xl then "text-9xl"
    else "unknown"

let renderFontSize fontSize =
    let example =
        Daisy.button.button [
            button.outline
            button.primary
            button.lg
            fontSize
            prop.text (str fontSize)
        ]

    let code =
        $"""Daisy.button.button
            [   button.outline
                button.primary
                button.lg
                {str fontSize}
                prop.text "This is {str fontSize}" ]"""

    let title = Html.text "Use of fontSize is rather straightforward."
    codedView title code example

[<ReactComponent>]
let FontSizeView () =
    React.fragment [
        for fontSize in fontSizeStyles do
            renderFontSize fontSize
    ]

File: src/docs/output/Pages/FontSize.js
import { ofArray } from "../fable_modules/fable-library-js.4.24.0/List.js";
import { createObj, equals } from "../fable_modules/fable-library-js.4.24.0/Util.js";
import { createElement } from "react";
import React from "react";
import * as react from "react";
import { Helpers_combineClasses } from "../fable_modules/Feliz.DaisyUI.5.0.0/DaisyUI.fs.js";
import { codedView } from "../SharedView.js";
import { map, delay, toList } from "../fable_modules/fable-library-js.4.24.0/Seq.js";

export const fontSizeStyles = ofArray([["className", "text-xs"], ["className", "text-sm"], ["className", "text-base"], ["className", "text-lg"], ["className", "text-xl"], ["className", "text-2xl"], ["className", "text-3xl"], ["className", "text-4xl"], ["className", "text-5xl"], ["className", "text-6xl"], ["className", "text-7xl"], ["className", "text-8xl"], ["className", "text-9xl"]]);

export function str(prop) {
    if (equals(prop, ["className", "text-xs"])) {
        return "text-xs";
    }
    else if (equals(prop, ["className", "text-sm"])) {
        return "text-sm";
    }
    else if (equals(prop, ["className", "text-base"])) {
        return "text-base";
    }
    else if (equals(prop, ["className", "text-lg"])) {
        return "text-lg";
    }
    else if (equals(prop, ["className", "text-xl"])) {
        return "text-xl";
    }
    else if (equals(prop, ["className", "text-2xl"])) {
        return "text-2xl";
    }
    else if (equals(prop, ["className", "text-3xl"])) {
        return "text-3xl";
    }
    else if (equals(prop, ["className", "text-4xl"])) {
        return "text-4xl";
    }
    else if (equals(prop, ["className", "text-5xl"])) {
        return "text-5xl";
    }
    else if (equals(prop, ["className", "text-6xl"])) {
        return "text-6xl";
    }
    else if (equals(prop, ["className", "text-7xl"])) {
        return "text-7xl";
    }
    else if (equals(prop, ["className", "text-8xl"])) {
        return "text-8xl";
    }
    else if (equals(prop, ["className", "text-9xl"])) {
        return "text-9xl";
    }
    else {
        return "unknown";
    }
}

export function renderFontSize(fontSize) {
    const example = createElement("button", createObj(Helpers_combineClasses("btn", ofArray([["className", "btn-outline"], ["className", "btn-primary"], ["className", "btn-lg"], fontSize, ["children", str(fontSize)]]))));
    const code = `Daisy.button.button
            [   button.outline
                button.primary
                button.lg
                ${str(fontSize)}
                prop.text "This is ${str(fontSize)}" ]`;
    const title = "Use of fontSize is rather straightforward.";
    return codedView(title, code, example);
}

export function FontSizeView() {
    const xs = toList(delay(() => map(renderFontSize, fontSizeStyles)));
    return react.createElement(react.Fragment, {}, ...xs);
}

